pkg/distributor/query.go

```diff
@@ -308,7 +312,9 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSet ri
 		Timeseries:  make([]mimirpb.TimeSeries, 0, len(hashToTimeSeries)),
 	}
 	for _, series := range hashToChunkseries {
-		resp.Chunkseries = append(resp.Chunkseries, series)
+		// Make a deep copy of each TimeSeriesChunk object, as those
+		// could be reused after invoking ingester_client.ReuseQueryStreamResponse.
+		resp.Chunkseries = append(resp.Chunkseries, ingester_client.CopyTimeSeriesChunk(series))
```

Comment:
Reading the code I couldn't figure out where the optimization comes from given we're making a copy of the `TimeSeriesChunk`. The optimization is given by the fact that `accumulateChunks()` (called previously) does a deduplication of identical chunks. Thanks to the RF=3, many chunks are perfectly identical and so the optimization is effective because we only copy `TimeSeriesChunk` after having deduplicated it.

Please update the comment to clearly explain how this optimization works.